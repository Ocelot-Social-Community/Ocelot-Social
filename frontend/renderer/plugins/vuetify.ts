// eslint-disable-next-line import/no-unassigned-import
import '@mdi/font/css/materialdesignicons.css'
// eslint-disable-next-line import/no-unassigned-import
import 'vuetify/styles'
import { I18n, useI18n } from 'vue-i18n'
import { createVuetify } from 'vuetify'
import { createVueI18nAdapter } from 'vuetify/locale/adapters/vue-i18n'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default (i18n: I18n<any, NonNullable<unknown>, NonNullable<unknown>, string, false>) =>
  createVuetify({
    locale: {
      adapter: createVueI18nAdapter({ i18n, useI18n }),
    },
    ssr: true,
  })
