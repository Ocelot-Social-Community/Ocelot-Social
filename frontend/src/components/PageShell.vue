<template>
  <v-app>
    <TopMenu />
    <slot />
  </v-app>
</template>

<script lang="ts" setup>
import TopMenu from '#components/menu/TopMenu.vue'
</script>
