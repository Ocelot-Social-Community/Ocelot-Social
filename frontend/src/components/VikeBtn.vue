<template>
  <v-btn
    :variant="isRouteSelected($attrs.href as string) ? 'tonal' : 'flat'"
    @click.prevent="onClick($attrs.href as string)"
  >
    <slot />
  </v-btn>
</template>
<script lang="ts" setup>
import { navigate } from 'vike/client/router'

import { usePageContext } from '#context/usePageContext'

const pageContext = usePageContext()

function onClick(href: string) {
  return navigate(href)
}

const isRouteSelected = (href: string) => {
  if (href === '/app') {
    return pageContext.urlPathname.indexOf(href) === 0
  }
  return pageContext.urlPathname === href
}
</script>
