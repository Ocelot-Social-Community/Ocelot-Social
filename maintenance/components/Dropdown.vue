<template>
  <v-popover
    v-model:open="isPopoverOpen"
    :open-group="Math.random().toString()"
    :placement="placement"
    :disabled="disabled"
    trigger="manual"
    :offset="offset"
  >
    <slot :toggle-menu="toggleMenu" :open-menu="openMenu" :close-menu="closeMenu" :is-open="isOpen" />
    <template #popover>
<div  @mouseover="popoverMouseEnter" @mouseleave="popoverMouseLeave">
      <slot
        name="popover"
        :toggle-menu="toggleMenu"
        :open-menu="openMenu"
        :close-menu="closeMenu"
        :is-open="isOpen"
      />
    </div>
</template>
  </v-popover>
</template>

<script>
let mouseEnterTimer = null
let mouseLeaveTimer = null

export default {
  props: {
    placement: { type: String, default: 'bottom-end' },
    disabled: { type: Boolean, default: false },
    offset: { type: [String, Number], default: '16' },
  },
  data() {
    return {
      isPopoverOpen: false,
      developerNoAutoClosing: false, // stops automatic closing of menu for developer purposes: default is 'false'
    }
  },
  computed: {
    isOpen() {
      return this.isPopoverOpen
    },
  },
  watch: {
    isPopoverOpen: {
      immediate: true,
      handler(isOpen) {
        try {
          if (isOpen) {
            this.$nextTick(() => {
              setTimeout(() => {
                const paddingRightStyle = `${
                  window.innerWidth - document.documentElement.clientWidth
                }px`
                const navigationElement = document.querySelector('.main-navigation')
                document.body.style.paddingRight = paddingRightStyle
                document.body.classList.add('dropdown-open')
                if (navigationElement) {
                  navigationElement.style.paddingRight = paddingRightStyle
                }
              }, 20)
            })
          } else {
            const navigationElement = document.querySelector('.main-navigation')
            document.body.style.paddingRight = null
            document.body.classList.remove('dropdown-open')
            if (navigationElement) {
              navigationElement.style.paddingRight = null
            }
          }
        } catch (err) {}
      },
    },
  },
  beforeUnmount() {
    clearTimeout(mouseEnterTimer)
    clearTimeout(mouseLeaveTimer)
  },
  methods: {
    toggleMenu() {
      this.isPopoverOpen ? this.closeMenu(false) : this.openMenu(false)
    },
    openMenu(useTimeout) {
      if (this.disabled) {
        return
      }
      this.clearTimeouts()
      if (useTimeout === true) {
        this.popoverMouseEnter()
      } else {
        this.isPopoverOpen = true
      }
    },
    closeMenu(useTimeout) {
      if (this.disabled) {
        return
      }
      this.clearTimeouts()
      if (useTimeout === true) {
        this.popoverMouseLeave()
      } else {
        this.isPopoverOpen = false
      }
    },
    popoverMouseEnter() {
      if (this.disabled) {
        return
      }
      this.clearTimeouts()
      if (!this.isPopoverOpen) {
        mouseEnterTimer = setTimeout(() => {
          this.isPopoverOpen = true
        }, 500)
      }
    },
    popoverMouseLeave() {
      if (this.developerNoAutoClosing) return
      if (this.disabled) {
        return
      }
      this.clearTimeouts()
      if (this.isPopoverOpen) {
        mouseLeaveTimer = setTimeout(() => {
          this.isPopoverOpen = false
        }, 300)
      }
    },
    clearTimeouts() {
      clearTimeout(mouseEnterTimer)
      clearTimeout(mouseLeaveTimer)
    },
  },
}
</script>
