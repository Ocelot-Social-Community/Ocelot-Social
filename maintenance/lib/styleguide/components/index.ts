import Avatar from './data-display/Avatar/Avatar.vue'
import CopyField from './data-display/CopyField/CopyField.vue'
import List from './data-display/List/List.vue'
import Number from './data-display/Number/Number.vue'
import Table from './data-display/Table/Table.vue'
import Form from './data-input/Form/Form.vue'
import FormItem from './data-input/FormItem/FormItem.vue'
import Input from './data-input/Input/Input.vue'
import Radio from './data-input/Radio/Radio.vue'
import Select from './data-input/Select/Select.vue'
import Container from './layout/Container/Container.vue'
import Card from './layout/Card/Card.vue'
import Grid from './layout/Grid/Grid.vue'
import Modal from './layout/Modal/Modal.vue'
import Page from './layout/Page/Page.vue'
import PageTitle from './layout/PageTitle/PageTitle.vue'
import Placeholder from './layout/Placeholder/Placeholder.vue'
import Section from './layout/Section/Section.vue'
import Spinner from './layout/Spinner/Spinner.vue'
import Button from './navigation/Button/Button.vue'
import Menu from './navigation/Menu/Menu.vue'
import Chip from './typography/Chip/Chip.vue'
import Code from './typography/Code/Code.vue'
import Heading from './typography/Heading/Heading.vue'
import Icon from './typography/Icon/Icon.vue'
import Logo from './typography/Logo/Logo.vue'
import Tag from './typography/Tag/Tag.vue'
import Text from './typography/Text/Text.vue'

export {
  Avatar,
  CopyField,
  List,
  Number,
  Table,
  Form,
  FormItem,
  Input,
  Radio,
  Select,
  Container,
  Card,
  Grid,
  Modal,
  Page,
  PageTitle,
  Placeholder,
  Section,
  Spinner,
  Button,
  Menu,
  Chip,
  Code,
  Heading,
  Icon,
  Logo,
  Tag,
  Text,
}
