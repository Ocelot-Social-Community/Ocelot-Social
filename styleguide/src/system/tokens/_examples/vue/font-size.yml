#
# FONT-SIZE TOKENS
# Use these tokens to set a font-size.
#

props:
  - name: font-size-xxxx-large
    value: "3rem"
  - name: font-size-xxx-large
    value: "2.5rem"
  - name: font-size-xx-large
    value: "2rem"
  - name: font-size-x-large
    value: "1.5rem"
  - name: font-size-large
    value: "1.25rem"
  - name: font-size-base
    value: "1rem"
  - name: font-size-body
    value: "15px"
  - name: font-size-small
    value: "0.8rem"
  - name: font-size-x-small
    value: "0.7rem"
  - name: font-size-xx-small
    value: "0.6rem"
global:
  type: number
  category: font-size
