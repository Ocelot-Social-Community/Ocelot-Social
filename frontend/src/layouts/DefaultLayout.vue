<template>
  <v-main class="bg-grey-lighten-3">
    <v-container>
      <v-row>
        <v-col v-if="slots.sidemenu" cols="2">
          <v-sheet rounded>
            <slot name="sidemenu"></slot>
          </v-sheet>
        </v-col>

        <v-col>
          <v-sheet rounded class="pa-3">
            <slot />
          </v-sheet>
        </v-col>
      </v-row>
    </v-container>
  </v-main>
</template>

<script lang="ts" setup>
import { useSlots } from 'vue'

const slots = useSlots()
</script>
