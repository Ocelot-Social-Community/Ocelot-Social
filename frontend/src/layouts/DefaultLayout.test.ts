import { mount } from '@vue/test-utils'
import { beforeEach, expect, describe, it } from 'vitest'
import { Component, h } from 'vue'
import { VApp } from 'vuetify/components'

import DefaultLayout from './DefaultLayout.vue'

describe('LogoAvatar', () => {
  const Wrapper = () => {
    return mount(VApp, {
      slots: {
        default: h(DefaultLayout as Component),
      },
    })
  }
  let wrapper: ReturnType<typeof Wrapper>

  beforeEach(() => {
    wrapper = Wrapper()
  })

  it('renders', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
