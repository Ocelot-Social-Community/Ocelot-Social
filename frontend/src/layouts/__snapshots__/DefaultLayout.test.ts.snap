// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LogoAvatar > renders 1`] = `
<div
  class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr"
>
  <div
    class="v-application__wrap"
  >
    
    <main
      class="v-main bg-grey-lighten-3"
      style="--v-layout-left: 0px; --v-layout-right: 0px; --v-layout-top: 0px; --v-layout-bottom: 0px; transition: none !important;"
    >
      
      <div
        class="v-container v-locale--is-ltr"
      >
        <div
          class="v-row"
        >
          <!--v-if-->
          <div
            class="v-col"
          >
            <div
              class="v-sheet v-theme--light v-sheet--rounded pa-3"
            >
              
              
            </div>
          </div>
        </div>
      </div>
      
    </main>
    
  </div>
</div>
`;
